rfcs/convergence/apperance-migration.md

```diff
@@ -0,0 +1,106 @@
+
+## Problem statement
+
+This RFC explores potential solutions for partners, so they could migrate input components to V9 without the extra work of adding an additional prop to input components.
+
+Partners should also easily revert the decision to the default appearance value if they decide so, without changing inputs individually.
+
+👎 Cons of adding appearance to achieve the same design as V0:
+
+- Not scalable, partners would have to add the prop to every component
+- If they decide to revert back to default appearance in future, they would have to go through the same pain again by removing the prop
+- Is error prone
+
+## Detailed Design or Proposal
+
+### Composed component
+
+Partners could create a new composed component in fluent and modify the props in their preferred way. If the partner would like to keep the original color, they could create the composed component and have the default component without the appearance prop renders as `filled-darker`.
+
+👍 Pros:
+
+- Is relatively safe
+
+👎 Cons:
+
+- Creating new composed component for each input `Dropdown`, `Input`, `TextArea` and `DatePicker`
+- Composed component apperance prop wouldn't match our Fluent V9 documentation
+- Does not work if repos have dependencies of another project with Fluent V9 input components (Nova)
+- Composition approach is currently marked as unstable in v9
+
+### Wrap the library components on application side
+
+Partners could a new component wrapper that will wrap input components and modify the appearance prop on the application side.
+
+👍 Pros:
+
+- Is safe and stable
+
+👎 Cons:
+
+- Creating new composed component for each input `Dropdown`, `Input`, `TextArea` and `DatePicker`
+- Wrapper component apperance wouldn't match our Fluent V9 documentation
+- Would work in iframes only within the TMP repository
+- Does not work if repos have dependencies of another project with Fluent V9 input components (Nova)
+
+### Global css selector
+
+Targeting all input selectors from a partner app and change the color with global css.
+
+Example: https://codesandbox.io/s/rfc-inputs-ew821q?file=/app.tsx
+
+👍 Pros:
+
+- Relatively easy and fast to do
+- Scalable
+- Would work for dependencies such as Nova (but depending where we inject the CSS rule and if they have their own `FluentProvider`)
+
+👎 Cons:
+
+- Difficult to validate
+
+### New token alias to theme
+
+Adding a new alias color token (let's say `colorInputBackground`) and use it for all inputs.
+
+👍 Pros:
+
+- Easy to do and would work for all inputs
+
+👎 Cons:
+
+- Negative impact on performance by increasing variables (as read here: [fluentui/theme-shared-colors.md at d5d510bf1ffcc1a4ed2067e9eb009c84e7beb351 · microsoft/fluentui (github.com)](https://github.com/microsoft/fluentui/blob/d5d510bf1ffcc1a4ed2067e9eb009c84e7beb351/rfcs/react-components/convergence/theme-shared-colors.md))
```

Comment:
I think what you are proposing is similar to the "Another option" bellow.
The question is how we know what style attribute requires this "extension point"?
If we have too many of them it might impact perf and make development more expensive (we should test or permutations of these optional tokens work as expected.

So should we add them only when needed? Like saying in the design spec that `Input` background should be customizable through the optional variable?

Reply:
Yeah, it's essentially the same as 'Another Option'. The one addition we could consider is somehow actually adding this to a theme, or some kind of theme extension, this would help with the discoverability challenge. This makes more sense for something like responsive breakpoints, where you could imagine all of them being defined in an optional theme extension, but perhaps makes a little less sense for this single extension point.